### PR TITLE
Services finder

### PR DESCRIPTION
**Pull request by roarking**
@vonnagy This PR would give anybody using the server the ability to look up services from the container itself.  Right now, we can register services (actors) with the container, but those services are isolated from that point on.

Right now, the only way to get a reference to these services is by using ActorSelection, but that implies knowing the exact path where the service is running.

Since we are providing a name as part of the container build process, I believe there should be a way to get an actual reference to these services using this same name, so that RoutedEndpoints or Listeners can send messages directly to an ActorRef.

I added a method called findService(name:String) that returns a Future[Option[ActorRef]] that does that.

I also modified some of the mutability/side effects going on in the ContainerService class by creating a lazy val applicativ to initialize the services.

Let me know what you think!

Thanks,

-Alex

### CODE REVIEW

**Comment by vonnagy** [comment]
@roarking I really like the idea here. A questions I have is that the findService() method is at the ContainerService level so it is only available when constructing the service. This method sends a message to the ServicesManager to find the appropriate service.

Would it be better suited to move the function to the ServicesManager object and thus allow static access to the method from any actor, routed endpoint, etc..? It would look something like the following:
```scala
def findService(name: String)(implicit system: ActorSystem): Future[Option[ActorRef]] = {
    implicit val timeout = Timeout(1 second)
    (system.actorSelection("akka://server/user/service") ? FindService(name)).mapTo[Option[ActorRef]]
  }
```

This way anyone can call is as we always know the static path to the services manager.

Let me know what you think.

Thanks,

Ivan

**Comment by roarking** [comment]
I think that's a great idea. Will commit these changes soon!

**Comment by roarking** [comment]
@vonnagy check it out please - I think this is what you are talking about.

File: service-container/src/test/scala/com/github/vonnagy/service/container/service/ServicesManagerSpec.scala [hunk 1]
```diff
@@ -52,9 +57,10 @@ class ServicesManagerSpec extends AkkaTestkitSpecs2Support(ActorSystem("test", {
 
       // Hijack all of the messages
       val intercept: PartialFunction[Any, Any] = {
-        case m @ HttpStopped =>
+        case m@HttpStopped =>
           act.underlyingActor.log.info(s"Intercepted HttpStopped message")
-          probe.ref ! m; m
+          probe.ref ! m;
```
Comment (vonnagy) [inline service-container/src/test/scala/com/github/vonnagy/service/container/service/ServicesManagerSpec.scala]: @roarking Remove the unused semi-colon and I think we can then merge the PR

Comment (roarking) [reply service-container/src/test/scala/com/github/vonnagy/service/container/service/ServicesManagerSpec.scala]: +1

**Comment by roarking** [comment]
Removed the extra semi-colon from the Spec class

**Pull request merged**